sql/parser/builtins.go

```diff
@@ -116,6 +117,170 @@ var builtins = map[string][]builtin{
 		},
 	},
 
+	"concat_ws": {
+		builtin{
+			fn: func(args DTuple) (Datum, error) {
+				dstr, ok := args[0].(DString)
+				if !ok {
+					return DNull, fmt.Errorf("unknown signature for concat_ws: concat_ws(%s, ...)", args[0].Type())
+				}
+				sep := string(dstr)
+				var ss []string
+				for _, d := range args[1:] {
+					if d == DNull {
+						continue
+					}
+					ds, err := datumToRawString(d)
+					if err != nil {
+						return DNull, err
+					}
+					ss = append(ss, ds)
+				}
+				return DString(strings.Join(ss, sep)), nil
+			},
+		},
+	},
+
+	"split_part": {
+		builtin{
+			types: typeList{stringType, stringType, intType},
+			fn: func(args DTuple) (Datum, error) {
+				text := string(args[0].(DString))
+				sep := string(args[1].(DString))
+				field := int(args[2].(DInt))
+
+				if field <= 0 {
+					return DNull, fmt.Errorf("field position must be greater than zero")
+				}
+
+				splits := strings.Split(text, sep)
+				if field > len(splits) {
+					return DString(""), nil
+				}
+				return DString(splits[field-1]), nil
+			},
+		},
+	},
+
+	"repeat": {
+		builtin{
+			types: typeList{stringType, intType},
+			fn: func(args DTuple) (Datum, error) {
+				s := string(args[0].(DString))
+				count := int(args[1].(DInt))
+				if count < 0 {
+					count = 0
+				}
+				return DString(strings.Repeat(s, count)), nil
+			},
+		},
+	},
+
+	"ascii": {stringBuiltin(1, func(s []string) (Datum, error) {
+		return DInt(int(s[0][0])), nil
```

Comment:
This will panic if `len(s[0]) == 0`. The docs also say: `For UTF8 returns the Unicode code point of the character`. So I think this should be something like:

```
  for _, ch := range s[0] {
    return DInt(ch), nil
  }
  return nil, fmt.Errorf("some error about the empty string")
```